src/spec.h

```diff
@@ -238,50 +238,52 @@ typedef struct {
 
 //---------------------------------------------------------------------------------------------
 
+
 typedef struct IndexSpec {
-  char *name;
-  FieldSpec *fields;
-  int numFields;
+  char *name;                     // Index name
+  uint64_t uniqueId;              // Id of index
+  FieldSpec *fields;              // Fields in the index schema
+  int numFields;                  // Number of fields
+
+  IndexStats stats;               // Statistics of memory used and quantities 
+  IndexFlags flags;               // Flags                   
 
-  IndexStats stats;
-  IndexFlags flags;
+  Trie *terms;                    // Trie of all terms. Used for GC and fuzzy queries
+  Trie *suffix;                   // Trie of suffix tokens of terms. Used for contains queries
+  t_fieldMask suffixMask;         // Mask of all field that support contains query
+  dict *keysDict;                 // Global dictionary. Contains inverted indexes of all TEXT terms
 
-  Trie *terms;
+  RSSortingTable *sortables;      // Contains sortable data of documents
 
-  RSSortingTable *sortables;
+  DocTable docs;                  // Contains metadata of all documents
 
-  DocTable docs;
+  StopWordList *stopwords;        // List of stopwords for TEXT fields
 
-  StopWordList *stopwords;
+  GCContext *gc;                  // Garbage collection
 
-  GCContext *gc;
+  SynonymMap *smap;               // List of synonym
+  char **aliases;                 // Aliases to self-remove when the index is deleted
```

Comment:
**If** we consider existing user modules built with current version, this could be a breaking change? (reordering of fields, adding new fields not only as last)

Reply:
These are the internals of RediSearch. It isn't exposed in the API.
@MeirShpilraien, do you see an issue here?